sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -1154,10 +1154,24 @@ mod checked {
                 FunctionKind::NonEntry
             }
             (Visibility::Private | Visibility::Friend, false) => {
-                return Err(ExecutionError::new_with_source(
-                    ExecutionErrorKind::NonEntryFunctionInvoked,
-                    "Can only call `entry` or `public` functions",
-                ));
+                if cfg!(msim) {
```

Comment:
This is not going to work with the adapter rewrite as is. 
Easiest idea I see right now: You could add a protocol config flag for these, with no intention of ever turning it on for any network. Then set the flag in the `init` section of the test